orchestrator/utils/state.py

```diff
@@ -150,8 +150,14 @@ def _build_arguments(func: StepFunc | InputStepFunc, state: State) -> list:  # n
 
     Raises:
          KeyError: if requested argument is not in the state, or cannot be reconstructed as an initial domain model.
+         ValueError: if requested argument cannot be converted to the expected type.
 
     """
+
+    def _convert_to_uuid(v: Any) -> UUID:
+        """Converts the value to a UUID instance if it is not already one."""
+        return v if isinstance(v, UUID) else UUID(v)
+
     sig = inspect.signature(func)
     arguments: list[Any] = []
     if sig.parameters:
```

Comment:
Maybe you can add an early exit here to reduce indentation?

```suggestion
    if not sig.parameters:
        return []
```